src/lib.rs

```diff
@@ -78,6 +102,23 @@ fn test_NumberOrString() {
     );
 }
 
+
+pub trait UriSource {
+    fn get_uri(&self) -> &Url;
+    
+    fn parse_file_path(&self) -> Result<PathBuf, Box<Error>> {
+        let uri = self.get_uri();
+        
+        if uri.scheme() != "file" {
+            return Err("URI scheme is not `file`".into());
+        }
+        
+        uri.to_file_path().map_err(|_err| "Invalid file path in URI".into())
+    }
+}
```

Comment:
Might as well remove `UrlSource` now? It does not seem to carry its own weight now `obj.url.to_file_path()` vs `obj.parse_file_path()`